### PR TITLE
Add Datadog Incident Management plugin support

### PR DESCRIPTION
**Pull request by bernardjkim**
This PR adds support for self-hosted Access Request Datadog plugin.

I'll create a separate PR for a few todo items that are also required to support the plugin.
Todo:
- [ ] https://github.com/gravitational/teleport.e/pull/4992
- [ ] https://github.com/gravitational/teleport/pull/46514
- [ ] Add support for automatic approvals through Datadog oncall

changelog: Add support for Access Request Datadog plugin.

### CODE REVIEW

**Comment by zmb3** [comment]
I woudl prefer that we named this plugin something like `DatadogIncidentManagement` as this is a separate product (similar to PagerDuty), not Datadog's traditional monitoring solution (which there is also a Teleport integration for)

File: api/proto/teleport/legacy/types/types.proto
Comment (zmb3) [inline api/proto/teleport/legacy/types/types.proto:6053]:
```suggestion
    // Settings for the Datadog Incident Management plugin
```

**Comment by tigrato** [comment]
Is it possible to split this PR into smaller PRs?

**Comment by bernardjkim** [comment]
Friendly ping @hugoShaka @EdwardDowling @marcoandredinis @tigrato

File: integrations/access/datadog/cmd/teleport-datadog/main.go
Comment (marcoandredinis) [inline integrations/access/datadog/cmd/teleport-datadog/main.go:93]: What's the 2nd argument? Can we extract the value into a more meaningful variable name?

File: integrations/access/datadog/client.go
Comment (marcoandredinis) [inline integrations/access/datadog/client.go:44]: Can we please add a godoc explaining what this permission allows?
A short sentence and link to https://docs.datadoghq.com/account_management/rbac/permissions/#case-and-incident-management should be enough.

File: integrations/access/datadog/client.go
Comment (marcoandredinis) [inline integrations/access/datadog/client.go:121]: Just a nit, but I think we can break the loop as soon as we find the permission we are looking for.

```suggestion
		if permission.Attributes.Name == IncidentWritePermissions {
			if permission.Attributes.Restricted {
				return trace.AccessDenied("missing incident_write permissions")
			}

			return nil
		}
```

**Review by marcoandredinis — COMMENTED** [review]
Most things look good
I want to do a real test before giving the final approval

File: integrations/access/datadog/client.go
Comment (hugoShaka) [reply integrations/access/datadog/client.go:186]: Historical context: for azure/msteams we did not use the official API because the SDK depends on the entire universe and it took 30 min + to build the plugin (I tried doing it but gave up).

For other implementations this was done by the martians and they likely walked the easiest/quickest pah.

I don't have an opinion on whether we should use or not their go client. I think the most dangerous things with building our own client are the authentication flow and the failure/retry logic. If those are trivial and the go client adds many deps, rolling our own client makes sense.

In this case maybe drop a comment saying why we decided to do it ourselves.

Comment (bernardjkim) [reply integrations/access/datadog/client.go:189]: I'll leave a todo to migrate to the official datadog package if the need arises.

**Review by marcoandredinis — APPROVED** [review]
Did a small test and things look good 👍 
image
image

File: integrations/access/datadog/types.go
Comment (tigrato) [inline integrations/access/datadog/types.go:23]: can you add documentation and describe where you got the information for these structures?

**Comment by bernardjkim** [comment]
Hey @tigrato @hugoShaka, are there any other major concerns with this PR? I will be make a couple follow up PRs, so we can always catch some minor issues then.

**Comment by hugoShaka** [comment]
Yeah, the Enterprise tests are not passing and I'm trying to figure out why. (They are not run in the CI because the plugin is in Teleport OSS)

**Review by hugoShaka — APPROVED** [review]

**Review by r0mant — APPROVED** [review]

**Pull request merged**
